Feature/check session storage

matsu0228 opened the pull request:
ルームidが自分のuser情報と異なる部屋に入っている疑惑（https://github.com/iot-social-network-link/meetalk/issues/73 ）への対応。

◆確認内容
①自分のユーザidを確認(画像では5)
②sessionStorageに格納されているpeerIDを確認
③sessionStorageクリア後のsessionStorage値を確認
④３でクリアしたことで、自分のuseidと同じpeerIDで入室できていること(画像では5)
　→本対応前では、sessionStorage値②を用いて入室していたみたい
![meetalk 2](https://cloud.githubusercontent.com/assets/5501329/12213929/4a64a68c-b6c6-11e5-91ff-53ecd5eaba89.png)

matsu0228 committed:
sessionStorageのlogging、clear
logging用の関数追加
URLにhttpsを含める

matsu0228 committed:
bugfix

xkumiyu commented:
- sessionStorageをクリアする必要ある？
  - 既に格納されてたらその値を使うとかにできない？
  - ブラウザの更新したとき、クリアされて別の部屋にはいるとかにならない？
- top_urlは相対パスではできない？
  - もしくは現在のパスを取得するとかで定数にはしないほうがいいいと思う。
  - 自動テストのときはポート違うから上手くいかない。
- logging_debugってcondole.logとは違うの？

yuichikubota commented:
> 既に格納されてたらその値を使うとかにできない？

それすると保持されたwindow id使って前回のroomに接続されることになるから、こっちで明示的にクリアした方が良いと思う。

> ブラウザの更新したとき、クリアされて別の部屋にはいるとかにならない？

その可能性あるね。ローカルで試してみるけど。
Exitボタン押されたタイミングとか、トーク終了時にクリアするとか？
ユーザの挙動に対する処理をディスカッションして決めよう。

> logging_debugってcondole.logとは違うの？

フラグで表示を切り替えるために関数作ったってことで合ってる？

matsu0228 committed:
TOP_URLの記述を変更

現在のホストから自動で取得するように変更。
そのうち、下記をsettings.ymlで定義する
・URI_SCHEME：　ローカルではhttp / 本番サーバではhttps
・DEBUG_FLG：　devではtrue / 本番ではfalse

matsu0228 commented:
> sessionStorageをクリアする必要ある？ 

ある。別の部屋に入ることはない。
前提説明してなくてごめんけど、下記のとおり。
- skywayに接続するときに、下記を初期値として渡す。
  - room(skyway側で部屋を識別するid): room_id
  - id: 'meetaikid_' + user_id
    (skyway側でユーザを識別するid。window id(=PeerID)は、経験則からこのidの先頭にランダム文字付加したものっぽい)
- skywayの仕様として、初回接続時に、sessionStorageにPeerIDを格納するらしい
  - ★重要★skyway仕様として、skywayの接続時にidを渡しても、sessionStorageがあれば優先して接続してしまう(画像例だとuid=4)
  - sessionStorageはhmlt5仕様より、タブを消すまでデータを保存できる
  - すなわち、一度退室して再度入室したときは、sessionStorageに保存済みユーザで接続する(画像例だとuid=4)こととなり、サーバ側で払いだされた(画像例だとuid=5)room_idと異なる部屋に入室しうる
- 別の部屋に入らない理由としては、ブラウザcookieの値通り、roomをskywayに渡せるため、その懸念はないはず

> TOP_URL
- 現在のパスを取得するに変えました(https://github.com/iot-social-network-link/meetalk/commit/39144fddf1f619bbfa70c4fe5ebb918ff72112cb)
- 下記をsettings.ymlで定義するようにしたいです。書き方わからないので、時間あれば確認しやす。できたら直してほしいです。。
  - URI_SCHEME：　ローカルではhttp / 本番サーバではhttps
  - DEBUG_FLG：　devではtrue / 本番ではfalse

> logging_debug

くぼっちのいうとおり、「DEBUG_FLG」がtrueのときだけcosole.logする関数。
→productionでは、内部情報見せないようにフラグ管理したく。

ykubot committed:
SessionStorageが削除されることを確認。
リロード時にもクリアされる件については、現在skyway側にroom id、user idを引き渡しており、リロードでクリアしても結局同じwindow idが振られる実装になっているため問題なし。
SessionStorageを関数にして外出しにした。
